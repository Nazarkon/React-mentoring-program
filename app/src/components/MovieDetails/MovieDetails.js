import React, { useEffect, useState } from 'react';
import { useParams } from 'react-router-dom';
import { getMovieListById } from '../../api/movie/controller';

import './MovieDetails.scss';
import { createMovieItemsList } from '../../helpers/MovieItemsHelpers';

function MovieDetails() {
  const [movieInfo, setMovieInfo] = useState();

  const { id } = useParams();

  useEffect(() => {
    getMovieListById(id)
      .then((data) => {
        setMovieInfo(data);
      })
      .catch((error) => {
        console.error(error);
      });
  }, [id]);

  return (
    <div className="movie-details-container">
      <div className="movie-details-container-image">
        <img className="movie-card-image" alt="film poster" src={movieInfo?.imageUrl} />
      </div>
      <div className="movie-details-container-description">
        <div className="movie-details-description-title">
          <h4 className="movie-details-name">{movieInfo?.name}</h4>
          <span className="movie-details-rating">{movieInfo?.rating}</span>
        </div>
        <span className="movie-details-genres">{createMovieItemsList(movieInfo?.genreList)}</span>
        <div className="movie-details-container-subtitle">
          <span className="movie-details-year">{movieInfo?.year}</span>
          <span className="movie-details-duration">{movieInfo?.duration}</span>
        </div>
        <p className="movie-details-description">{movieInfo?.description}</p>
      </div>
    </div>
  );
}

// MovieDetails.propTypes = {
//   movieInfo: PropTypes.object.isRequired
// };
export default MovieDetails;
