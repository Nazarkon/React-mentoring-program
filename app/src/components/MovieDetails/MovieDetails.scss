.movie-details-container{
    width: 100%;
    max-width: 1200px;
    background: #424242;
    display: flex;
        .movie-details-container-image{
            width: 100%;
            min-width: 400px;
                .movie-card-image{
                    width: 100%;
                    height: 100%;
                    object-fit: cover;
                }
        }
        .movie-details-container-description{
            margin:0px 50px 0px 60px;
            .movie-details-description-title{
                display: flex;
                justify-content: flex-start;
                align-items: center;
                    .movie-details-name{
                        font-size: 40px;
                        color: #FFFFFF;
                        line-height: 49px;
                        text-transform: uppercase;
                        margin-right: 29px;
                        flex: 6;
                    }
                    .movie-details-rating{
                        width: 60px;
                        height: 60px;
                        border: 1px solid #FFFFFF;
                        border-radius: 100%;
                        display: flex;
                        justify-content: center;
                        align-items: center;
                        text-transform: uppercase;
                        font-size: 20px;
                        line-height: 24px;
                        color: #FFFFFF;
                    }
            }
            .movie-details-genres{
                color: #FFFFFF;
                opacity: 0.5;
            }
            .movie-details-container-subtitle{
                color: #F65261;
                font-size: 24px;
                margin: 29px 0px 29px 0px;
                    .movie-details-year{
                        margin-right: 51px;
                    }
            }
            .movie-details-description{
                word-break: break-all;
                color:#FFFFFF;
                opacity: 0.5;
                font-weight: 300;
                font-size: 20px;
                line-height: 24px;
                text-align: left;
            }
        }
}

.movie-details-error-container {
    background-color: wheat;
    border-radius: 20px;
    width: 100%;
    max-width: 400px;
    padding: 100px 0px;
    margin: 0 auto;
    font-size: 70px;
    text-align: center;
    color:#424242
}