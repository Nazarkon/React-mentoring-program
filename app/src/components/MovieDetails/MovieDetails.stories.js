import React from 'react';

import MovieDetails from './MovieDetails';

import filmImage from '../../assets/images/Bitmap.png';

export default {
  title: 'Components/MovieDetails',
  component: MovieDetails,
  argTypes: {
    movieInfo: {
      control: 'object',
      description: 'Object that includes all information about film',
      defaultValue: {
        imageUrl: filmImage,
        name: 'Pulp Fiction',
        year: '2003',
        rating: '8.9',
        duration: '2h 34min',
        description:
          'Jules Winnfield (Samuel L. Jackson) and Vincent Vega (John Travolta) are two hit men who are out to retrieve a suitcase stolen from their employer, mob boss Marsellus Wallace (Ving Rhames). Wallace has also asked Vincent to take his wife Mia (Uma Thurman) out a few days later when Wallace himself will be out of town. Butch Coolidge (Bruce Willis) is an aging boxer who is paid by Wallace to lose his fight. The lives of these seemingly unrelated people are woven together comprising of a series of funny, bizarre and uncalled-for incidents.â€”Soumitra',
        genreList: ['Action', 'Adventure']
      }
    }
  }
};

const Template = (args) => <MovieDetails {...args} />;

export const Default = Template.bind({});
