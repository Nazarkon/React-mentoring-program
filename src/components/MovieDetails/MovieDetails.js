import React from "react";

import "./MovieDetails.scss"
import { createMovieItemsList } from '../../helpers/MovieItemsHelpers'

const MovieDetails = ({movieInfo}) => {

    if(!movieInfo || !Object.entries(movieInfo).length) {
        return <div className="movie-details-error-container">Not Found</div>
    }


    const { imageUrl, name, year, rating, duration, description, genreList } = movieInfo

    return (
        <div className="movie-details-container">
            <div className="movie-details-container-image">
                <img className="movie-card-image" alt="film poster" src={imageUrl}/>
            </div>
                <div className="movie-details-container-description">
                    <div className="movie-details-description-title">
                        <h4 className="movie-details-name">{name}</h4>
                        <span className="movie-details-rating">{rating}</span>
                    </div>
                        <span className="movie-details-genres">{createMovieItemsList(genreList)}</span>
                    <div className="movie-details-container-subtitle">
                        <span className="movie-details-year">{year}</span>
                        <span className="movie-details-duration">{duration}</span>
                    </div>
                    <p className="movie-details-description">{description}</p>
                </div>
        </div>
    )
}

export default MovieDetails