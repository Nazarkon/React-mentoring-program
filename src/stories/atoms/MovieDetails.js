import React from "react";

import "../assets/styles/MovieDetails.scss"

const MovieDetails = ({movieInfo}) => {

    const { imageUrl, name, year, rating, duration, description, genreList } = movieInfo

    const createListOfGenres = () => {
        if(!genreList || !genreList.length) return '-';

        return genreList.join('&');
    }

    return (
        <div className="movie-details-container">
            <div className="movie-details-container-image">
                <img className="movie-card-image" alt="film poster" src={imageUrl}/>
            </div>
                <div className="movie-details-container-description">
                    <div className="movie-details-description-title">
                        <h4 className="movie-details-name">{name}</h4>
                        <span className="movie-details-rating">{rating}</span>
                    </div>
                        <span className="movie-details-genres">{createListOfGenres()}</span>
                    <div className="movie-details-container-subtitle">
                        <span className="movie-details-year">{year}</span>
                        <span className="movie-details-duration">{duration}</span>
                    </div>
                    <p className="movie-details-description">{description}</p>
                </div>
        </div>
    )
}

export default MovieDetails