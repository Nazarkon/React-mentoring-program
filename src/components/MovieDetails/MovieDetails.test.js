/* eslint-disable testing-library/prefer-screen-queries */
import { render } from '@testing-library/react';
import '@testing-library/jest-dom'
import MovieDetails from "./MovieDetails";

describe("MovieDetails component", () => {

    const mockMovieInfo = {
        imageUrl: 'Bitmap.png',
        name: 'Pulp Fiction',
        year: '2003',
        rating: '8.9',
        duration: '2h 34min', 
        description: 'Jules Winnfield (Samuel L. Jackson) and Vincent Vega (John Travolta) are two hit men who are out to retrieve a suitcase stolen from their employer, mob boss Marsellus Wallace (Ving Rhames). Wallace has also asked Vincent to take his wife Mia (Uma Thurman) out a few days later when Wallace himself will be out of town. Butch Coolidge (Bruce Willis) is an aging boxer who is paid by Wallace to lose his fight. The lives of these seemingly unrelated people are woven together comprising of a series of funny, bizarre and uncalled-for incidents.â€”Soumitra', 
        genreList: ['Action', 'Adventure']
    }

    it("Matches snapshot", () => {
        const { container } = render(
        <MovieDetails 
            movieInfo={mockMovieInfo}
        />);
        expect(container).toMatchSnapshot();
      });
      it("Check that component renders movie information correctly", () => {
        const { getByAltText, getByText } = render(
            <MovieDetails 
            movieInfo={mockMovieInfo}
            />
        );

        expect(getByAltText("film poster")).toHaveAttribute(
            "src",
            "Bitmap.png"
          );
        
        expect(getByText("Pulp Fiction")).toBeInTheDocument();

        expect(getByText("2003")).toBeInTheDocument();

        expect(getByText("Action&Adventure")).toBeInTheDocument();
      })
})