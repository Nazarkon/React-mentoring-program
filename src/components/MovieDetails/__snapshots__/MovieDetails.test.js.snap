// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails component Matches snapshot 1`] = `
<div>
  <div
    class="movie-details-container"
  >
    <div
      class="movie-details-container-image"
    >
      <img
        alt="film poster"
        class="movie-card-image"
      />
    </div>
    <div
      class="movie-details-container-description"
    >
      <div
        class="movie-details-description-title"
      >
        <h4
          class="movie-details-name"
        />
        <span
          class="movie-details-rating"
        />
      </div>
      <span
        class="movie-details-genres"
      >
        -
      </span>
      <div
        class="movie-details-container-subtitle"
      >
        <span
          class="movie-details-year"
        />
        <span
          class="movie-details-duration"
        />
      </div>
      <p
        class="movie-details-description"
      />
    </div>
  </div>
</div>
`;
